common/djangoapps/student/models.py

```diff
@@ -1793,6 +1793,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user).exists():
+                # if user has not profile then create it.
+                user_profile = UserProfile(name=user.username, user=user)
+                user_profile.save()
+
```

Comment:
Should this `UserProfile` creation be outside the "`PREVENT_CONCURRENT_LOGINS` is `True`" logic?